documents/entity/bootstrapper.go

```diff
@@ -63,16 +64,29 @@ func (Bootstrapper) Bootstrap(ctx map[string]interface{}) error {
 		return errors.New("entity relation service not initialised")
 	}
 
-	processor, _ := ctx[documents.BootstrappedAnchorProcessor].(documents.DocumentRequestProcessor)
-	/*if !ok {
-		return errors.New("processor not initialised")
-	}*/
+	processorFinder := func() documents.DocumentRequestProcessor {
+		processor, ok := ctx[documents.BootstrappedAnchorProcessor].(documents.DocumentRequestProcessor)
+		if !ok {
+			panic("processor initialisation error")
```

Comment:
I am not super keen on this type of handling as they will fail as panic when the node is running...which can cause other unforseen inconsistencies on already running processes. Why we cannot check that in bootstrap?

Reply:
I adopted the pattern from the invoice bootstrapper with the NFT dependency. 

Currently, the processor is bootstrapped in the PostDocumentBoostrapper because the processor depends on p2p etc.

Reply:
Couldnt we have the entity.Bootstrapper{} invoked after the documents.PostDocumentBootstrapper{}?

Reply:
@vimukthi-git Introduced the closure idea.

In some way I like it, because we don't have to care about the dependencies anymore.

Reply:
Instead of trying to rearrange the order of the bootstrappers

Reply:
@mikiquantum as explained offline its only needed in specific situations, I think it should be fine to do it here, as there is a legitimate init cycle here :)